File: 백준/Platinum/20041. Escaping/Escaping.cc
#include <bits/stdc++.h>
using namespace std;
typedef pair<int, int> pi;

vector<pi> cops;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int N; cin >> N;
    cops.reserve(N);
    for (int i = 0; i < N; i++) {
        int xi, yi; cin >> xi >> yi;

        cops.emplace_back(xi, yi);
    }
    int sx, sy; cin >> sx >> sy;

    bool isNorth = true, isEast = true, isSouth = true, isWest = true;
    for (auto &[x, y] : cops) {
        int absDiffX = abs(x - sx);
        int absDiffY = abs(y - sy);

        bool flagX = false, flagY = false;

        if (absDiffX == absDiffY) {
            flagX = true;
            flagY = true;
        } else if (absDiffX > absDiffY) {
            flagX = true;
        } else {
            flagY = true;
        }

        if (flagX) {
            if (x > sx) {
                isEast = false;
            } else {
                isWest = false;
            }
        }
        if (flagY) {
            if (y > sy) {
                isNorth = false;
            } else {
                isSouth = false;
            }
        }
    }

    cout << ((isNorth || isEast || isSouth || isWest) ? "YES" : "NO");
    return 0;
}

File: 백준/Platinum/20041. Escaping/Escaping.py
import sys


def input():
    return sys.stdin.readline().rstrip()


def solution() -> str:
    N = int(input())

    cops = [tuple(map(int, input().split())) for i in range(N)]
    sx, sy = map(int, input().split())

    isNorth, isEast, isSouth, isWest = True, True, True, True
    for x, y in cops:
        absDiffX = abs(x - sx)
        absDiffY = abs(y - sy)
        if absDiffX == absDiffY:
            if x > sx:
                isEast = False
            else:
                isWest = False
            if y > sy:
                isNorth = False
            else:
                isSouth = False
        elif absDiffX > absDiffY:
            if x > sx:
                isEast = False
            else:
                isWest = False
        else:
            if y > sy:
                isNorth = False
            else:
                isSouth = False

    if isNorth or isEast or isSouth or isWest:
        return "YES"
    else:
        return "NO"


if __name__ == '__main__':
    print(solution())
